config/encrypt_file_test.go

```diff
@@ -0,0 +1,148 @@
+package config_test
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"io"
+
+	"code.cloudfoundry.org/cli/plugin"
+	"code.cloudfoundry.org/cli/plugin/pluginfakes"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pivotal-cf/spring-cloud-services-cli-plugin/config"
+	"github.com/pivotal-cf/spring-cloud-services-cli-plugin/httpclient"
+	"github.com/pivotal-cf/spring-cloud-services-cli-plugin/httpclient/httpclientfakes"
+	"github.com/pivotal-cf/spring-cloud-services-cli-plugin/test_support"
+)
+
+var _ = Describe("Encrypt file", func() {
+
+	const (
+		serviceRegistryInstance = "some-service-registry"
+		plainText               = "plain-text"
+		errorText               = "to err is human"
+		accessToken             = "access-token"
+		bearerAccessToken       = "bearer " + accessToken
+		serviceURI              = "service-uri/"
+		encryptURI              = "service-uri/encrypt"
+		cipherText              = "cipher-text"
+	)
+
+	var (
+		fakeCliConnection     *pluginfakes.FakeCliConnection
+		fakeAuthClient        *httpclientfakes.FakeAuthenticatedClient
+		fakeResolver          func(cliConnection plugin.CliConnection, serviceInstanceName string, accessToken string, authClient httpclient.AuthenticatedClient) (string, error)
+		resolverAccessToken   string
+		testError             error
+		postResponse          io.ReadCloser
+		postStatusCode        int
+		postErr               error
+		output                string
+		err                   error
+		accessTokenURI        string
+		fakeResolverCallCount int
+		fakeResolverError     error
+	)
+
+	BeforeEach(func() {
+		fakeCliConnection = &pluginfakes.FakeCliConnection{}
+		fakeAuthClient = &httpclientfakes.FakeAuthenticatedClient{}
+		testError = errors.New(errorText)
+		postResponse, postStatusCode, postErr = ioutil.NopCloser(bytes.NewBufferString(cipherText)), http.StatusOK, nil
+		accessTokenURI = "access-token-uri"
+		fakeCliConnection.AccessTokenReturns(bearerAccessToken, nil)
+		fakeResolverCallCount = 0
+		fakeResolver = func(cliConnection plugin.CliConnection, serviceInstanceName string, accessToken string, authClient httpclient.AuthenticatedClient) (string, error) {
+			fakeResolverCallCount++
+			resolverAccessToken = accessToken
+			return serviceURI, fakeResolverError
+		}
+		config.DefaultResolver = fakeResolver
+		fakeResolverError = nil
+	})
+
+	JustBeforeEach(func() {
+		fakeAuthClient.DoAuthenticatedPostReturns(postResponse, postStatusCode, postErr)
```

Comment:
This looks a bit odd. Usually, `JustBeforeEach` is used to drive the subject of the test, e.g. make the call to `Encrypt`. It can do more than that, but the idea is to avoid duplication. It can refer to variables which are set in `BeforeEach` blocks.

Reply:
This was "borrowed" in the copy-n-paste from encrypt_test.go. I kept it around because mainly I wasn't entirely sure what it was doing and also because if I removed it, then the tests would fail.

Reply:
Oh fair enough. Sorry for not realising that.